pkg/provider/github/github.go

```diff
@@ -281,6 +279,9 @@ func (v *Provider) SetClient(ctx context.Context, run *params.Run, event *info.E
 	if v.Client == nil {
 		v.Client = client
 	}
+	if v.Client == nil {
+		return fmt.Errorf("no github client has been initialized")
+	}
```

Comment:
Should we make sure this check for other provider as well ??

Reply:
i think we are doing it in other providers